res/res_pjsip_rfc3326.c

```diff
@@ -53,12 +53,17 @@ static void rfc3326_use_reason_header(struct ast_sip_session *session, struct pj
 		cause_q850 = !strncasecmp(cause, "Q.850", 5);
 		cause_sip = !strncasecmp(cause, "SIP", 3);
 		if ((cause_q850 || cause_sip) && (cause = strstr(cause, "cause="))) {
-			/* If text is present get rid of it */
+			/* If text is present, process and get rid of it */
 			if ((text = strchr(cause, ';'))) {
+				if (!strncasecmp(text, "text=\"Call completed elsewhere\"", 31)) {
+					code = AST_CAUSE_ANSWERED_ELSEWHERE;
+				}
 				*text = '\0';
 			}
 
-			if (sscanf(cause, "cause=%30d", &code) != 1) {
+			if (code) {
+				continue;
```

Comment:
This seems wrong. If `code` is set here I assume you want to fall through to the `ast_channel_hangupcause_set(...)` branches below.

Reply:
I didn't fix it. As the code is currently written it will not fall through.